index.bs

```diff
@@ -2874,13 +2888,47 @@ Dictionary {{AudioTimestamp}} Members</h5>
                 {{AudioRenderCapacity/update}} at {{AudioRenderCapacity}}, using
                 {{AudioRenderCapacityEvent}}, with the given update interval in
                 {{AudioRenderCapacityOptions}}.
+
+                <div algorithm="audiorendercapacity-start">
+                    When invoked, perform the following steps:
+
+                    1. Set {{[[update interval]]}} to the value of
+                        <code>options.{{AudioRenderCapacityOptions/updateInterval}}
+                        </code>.
+                    1. Set {{[[is running]]}} to <code>true</code>.
+                    1. <a href="https://html.spec.whatwg.org/multipage/media.html#queue-a-media-element-task">
+                        Queue a media element task</a> to run the metric 
+                        reporting loop.
+                </div>
             
             : <dfn>stop()</dfn>
             ::
                 Stops metric collection and analysis. It also stops dispatching
                 {{AudioRenderCapacity/update}} events.
+
+                <div algorithm="audiorendercapacity-stop">
+                    When invoked, perform the following steps:
+
+                    1. Set {{[[is running]]}} to <code>false<code>.
+                </div>
         </dl>
 
+        <h6 id="AudioRenderCapacity-metric-reporting-loop">
+        Metric Reporting Loop</h6>
+
+        This algorithm can be invoked by {{AudioRenderCapacity/start}} method
+        to report the performance metric repreatedly.
+
+        1. If {{[[is running]]}} is <code>false</code>, abort this algorithm.
```

Comment:
I think if `is running` is only checked here it could result in multiple Metric Reporting Loops running in parallel.

I think the following could cause two parallel loops.

```
start();
stop();
start();
```

Since `stop()` only sets the `is running` flag to `false` but doesn't stop the loop both loops started by calling `start()` could be kept alive. It seems to be possible that by the time the first loop checks `is running` it's true again as a result of calling `start()` for the second time.